depthai_demo.py

```diff
@@ -172,35 +180,41 @@ def fn(value):
                 useRectifiedRight=Previews.rectified_right.name in conf.args.show and (conf.getModelSource() != "rectified_right" or not conf.args.sync),
             )
 
-        enc_manager = EncodingManager(pm, conf.args.encode, conf.args.encode_output) if len(conf.args.encode) > 0 else None
+        enc_manager = None
+        if len(conf.args.encode) > 1:
+            enc_manager = EncodingManager(conf.args.encode, conf.args.encode_output)
+            enc_manager.create_encoders(pm)
 
     if len(conf.args.report) > 0:
         pm.create_system_logger()
 
     if conf.useNN:
-        nn_pipeline = nn_manager.create_nn_pipeline(pm.p, pm.nodes,
-            source=conf.getModelSource(), shaves=conf.shaves, use_sbb=conf.args.spatial_bounding_box and conf.useDepth,
-            minDepth=conf.args.min_depth, maxDepth=conf.args.max_depth, sbbScaleFactor=conf.args.sbb_scale_factor,
-            use_depth=conf.useDepth, flip_detection=conf.getModelSource() in ("rectified_left", "rectified_right") and not conf.args.stereo_lr_check,
-            full_fov=not conf.args.disable_full_fov_nn,
+        nn_pipeline = nn_manager.create_nn_pipeline(
+            pipeline=pm.pipeline, nodes=pm.nodes, source=conf.getModelSource(),
+            blob_path=blob_manager.getBlob(shaves=conf.shaves, openvino_version=nn_manager.openvino_version),
+            use_depth=conf.useDepth, minDepth=conf.args.min_depth, maxDepth=conf.args.max_depth,
+            sbbScaleFactor=conf.args.sbb_scale_factor, full_fov=not conf.args.disable_full_fov_nn,
+            flip_detection=conf.getModelSource() in ("rectified_left", "rectified_right") and not conf.args.stereo_lr_check,
         )
 
-        pm.create_nn(nn=nn_pipeline, sync=conf.args.sync, use_depth=conf.useDepth, xout_nn_input=Previews.nn_input.name in conf.args.show,
-                     xout_sbb=conf.args.spatial_bounding_box and conf.useDepth)
+        pm.add_nn(
+            nn=nn_pipeline, sync=conf.args.sync, xout_nn_input=Previews.nn_input.name in conf.args.show,
+            use_depth=conf.useDepth, xout_sbb=conf.args.spatial_bounding_box and conf.useDepth
+        )
 
     # Start pipeline
-    device.startPipeline(pm.p)
+    device.startPipeline(pm.pipeline)
     pm.create_default_queues(device)
-    nn_in = device.getInputQueue(nn_manager.input_name, maxSize=1, blocking=False) if not conf.useCamera and conf.useNN else None
-    nn_out = device.getOutputQueue(nn_manager.output_name, maxSize=1, blocking=False) if conf.useNN else None
+    if conf.useNN:
+        nn_manager.createQueues(device)
```

Comment:
Yes, and the `camelCase` I think is a go here - it's not following python standards but combines well with our API. Didn't want to include it in this release though as the size of it would be too big, and it's something we can do after initial release

Reply:
> and it's something we can do after initial release

Change the coding style you mean?

I'd advise to do it before the first release IMO

Reply:
@themarpe I finished the refactor and everything inside the SDK and demo script is now camelCase